snuba/clickhouse/astquery.py

```diff
@@ -118,23 +120,37 @@ def _format_query_impl(self) -> str:
         if self.__limit is not None:
             limit_clause = f"LIMIT {self.__limit} OFFSET {self.__offset}"
 
-        self.__formatted_query = " ".join(
-            [
-                c
-                for c in [
-                    select_clause,
-                    from_clause,
-                    array_join_clause,
-                    prewhere_clause,
-                    where_clause,
-                    group_clause,
-                    having_clause,
-                    order_clause,
-                    limitby_clause,
-                    limit_clause,
-                ]
-                if c
+        self.__sql_data_list = [
+            (k, v)
+            for k, v in [
+                ("select", select_clause),
+                ("from", from_clause),
+                ("join", array_join_clause),
```

Comment:
array_join and join in clickhouse are two different concepts. I would call this array_join. ArrayJoin in clickhouse is like Postgres UNNEST.